src/Integration.UnitTests/Binding/ProjectRuleSetWriterTests.cs

```diff
@@ -1,475 +1,457 @@
-        {
-            // Setup
-            var fs = new ConfigurableRuleSetGenerationFileSystem();
-            var testSubject = new ProjectRuleSetWriter(fs);
-
-            const string solutionRoot = @"X:\SolutionDir\";
-            string solutionRuleSetPath = Path.Combine(solutionRoot, @"Sonar\Sonar1.ruleset");
-            string projectRuleSetRoot = Path.Combine(solutionRoot, @"Project\");
-
-            string existingRuleSetFileName = @"ExistingSharedRuleSet.ruleset";
-            string existingRuleSetFullPath = Path.Combine(solutionRoot, existingRuleSetFileName);
-            string existingRuleSetPropValue = PathHelper.CalculateRelativePath(projectRuleSetRoot, existingRuleSetFullPath);
-
-            testSubject.AlreadyUpdatedExistingRuleSetPaths.Add(existingRuleSetFullPath);
-            fs.AddRuleSetFile(existingRuleSetFullPath, new RuleSet("test"));
-            long beforeTimestamp = fs.GetFileTimestamp(existingRuleSetFullPath);
-
-            // Act
-            string pathOutResult;
-            bool result = testSubject.TryUpdateExistingProjectRuleSet(solutionRuleSetPath, projectRuleSetRoot, existingRuleSetPropValue, out pathOutResult);
-
-            // Verify
-            Assert.IsTrue(result, "Expected to return true when trying to update already updated existing rule set");
-            Assert.AreEqual(existingRuleSetFullPath, pathOutResult, "Unexpected rule set path was returned");
-
-            long afterTimestamp = fs.GetFileTimestamp(existingRuleSetFullPath);
-            Assert.AreEqual(beforeTimestamp, afterTimestamp, "Rule set timestamp has changed; file was unexpectedly written to.");
-        }
-
-        #endregion
-
-        #region Helpers
-
-        private static void AddRuleSetInclusion(RuleSet parent, RuleSet child, bool useRelativePath)
-        {
-            string include = useRelativePath
-                ? PathHelper.CalculateRelativePath(parent.FilePath, child.FilePath)
-                : child.FilePath;
-            parent.RuleSetIncludes.Add(new RuleSetInclude(include, RuleAction.Default));
-        }
-
-        #endregion
-    }
-}
+//-----------------------------------------------------------------------
+// <copyright file="ProjectRuleSetWriterTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using SonarLint.VisualStudio.Integration.Binding;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Globalization;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
+{
+    [TestClass]
+    public class ProjectRuleSetWriterTests

```

Comment:
Minor changes in practice, most of it due to unused variables (some of which needed to be used and caused minor adjustments to tests)